Auth0Tests/RequestSpec.swift

```diff
@@ -1,43 +1,157 @@
-                expect(request.headers["foo"]) == "baz"
+            context("headers") {
+
+                it("should create a request with headers") {
+                    let request = Request(session: URLSession.shared, url: Url, method: "GET", handle: plainJson, headers: ["foo": "bar"], logger: nil, telemetry: Telemetry())
+                    expect(request.headers["foo"]) == "bar"
+                }
+
+                it("should create a new request with extra headers") {
+                    let request = Request(session: URLSession.shared, url: Url, method: "GET", handle: plainJson, logger: nil, telemetry: Telemetry()).headers(["foo": "bar"])
+                    expect(request.headers["foo"]) == "bar"
+                }
+
+                it("should merge extra headers with existing headers") {
+                    let request = Request(session: URLSession.shared, url: Url, method: "GET", handle: plainJson, headers: ["foo": "bar"], logger: nil, telemetry: Telemetry()).headers(["baz": "qux"])
+                    expect(request.headers["foo"]) == "bar"
+                    expect(request.headers["baz"]) == "qux"
+                }
+
+                it("should overwrite existing headers with extra headers") {
+                    let request = Request(session: URLSession.shared, url: Url, method: "GET", handle: plainJson, headers: ["foo": "bar"], logger: nil, telemetry: Telemetry()).headers(["foo": "baz"])
+                    expect(request.headers["foo"]) == "baz"
+                }
+
+                it("should create a new request and not mutate an existing request") {
+                    let request = Request(session: URLSession.shared, url: Url, method: "GET", handle: plainJson, headers: ["foo": "bar"], logger: nil, telemetry: Telemetry())
+                    expect(request.headers(["foo": "baz"]).headers["foo"]) == "baz"
+                    expect(request.headers["foo"]) == "bar"
+                }
+
             }
 
-            it("should create a new request and not mutate an existing request") {
-                let request = Request(session: URLSession.shared, url: url, method: "GET", handle: plainJson, headers: ["foo": "bar"], logger: nil, telemetry: Telemetry())
-                expect(request.headers(["foo": "baz"]).headers["foo"]) == "baz"
-                expect(request.headers["foo"]) == "bar"
+        }
+
+        if #available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *) {
+            describe("combine") {
+                var cancellables: Set<AnyCancellable> = []
+
+                afterEach {
+                    cancellables.removeAll()
+                }
+
+                it("should emit only one value") {
+                    stub(condition: isHost(Url.host!)) { _ in
+                        return authResponse(accessToken: AccessToken, idToken: IdToken)
```

Comment:
Could you use something a bit simpler than an `authResponse`? like `HTTPStubsResponse(jsonObject: [:], statusCode: 200)`

Then you don't have to create AT and IdT's for a generic Request test.

Reply:
Good catch, changed in https://github.com/auth0/Auth0.swift/pull/566/commits/3bf54a23e24bb33d7bea2ab32aa10e72a3043050.